packages/core/_src/stm/TMap/operations/put.ts

```diff
@@ -0,0 +1,82 @@
+import type { Journal } from "@effect/core/stm/STM/Journal"
+import { concreteTArray, InternalTArray } from "@effect/core/stm/TArray/operations/_internal/InternalTArray"
+import { TMap } from "@effect/core/stm/TMap/definition"
+import { concreteTMap } from "@effect/core/stm/TMap/operations/_internal/InternalTMap"
+
+const LOAD_FACTOR = .75 as const
+
+/**
+ * Stores new binding into the map.
+ *
+ * @tsplus fluent ets/TMap put
+ */
+export function put_<K, V>(self: TMap<K, V>, k: K, v: V): USTM<void> {
+  function resize(journal: Journal, buckets: TArray<List<Tuple<[K, V]>>>): void {
```

Comment:
I would extract this to a non-exported function in the global scope so that we are not re-creating this function every time `put` is called. You can pass any additional arguments required to the function that it may be accessing now because it's in scope.